### ISSUE TITLE
memory leaking if pass a callback function as parameter

### ISSUE THREAD

**Issue by terrywh** [CopernicaMarketingSoftware/PHP-CPP#273]
at first i thought my implementation is buggy, and after i comment all the logics, and left only an empty function it still using up memory:
``` php
<?php
function test_promise() {
	// this is leaking and will be stopped when reaching memory limit
	$promise = promise(function() {});
	// this is ok
	$promise = promise("a");
	unset($promise);
}

for($i=0;$i<10000000;++$i) {
	test_promise();
}
```

the `promise` is a function that define in c++ using php-cpp:

``` c++
// function
Php::Value promise(Php::Parameters &params)
{
	return nullptr;
}
// add to extension
extension.add<promise>("promise");
```

i'm using v2.0.0 / php v7.0.13.

@EmielBruijntjes

**Comment by sjinks** [comment]
Possible duplicate of #249

**Comment by sjinks** [comment]
Yes, looks like this one is fixed in 88721ab27f5fe72f6fc91b445ec1db39e5ebc592

Could you please confirm?
